### DISCUSSION TITLE
waline服务端更新更好的方式，api 脚本调用

### DISCUSSION THREAD

**Discussion by appotry** [walinejs/waline#1935] [Ideas]
如何更新
进入到 GitHub 仓库中，修改 package.json 文件中的 @waline/vercel 版本号为最新版本即可。

文档中是这么写的，其实直接指定latest版本，重新部署一下服务端就更新到最新了。

开启一下vercel的自动部署api，使用curl调用一下就更新到最新了。完全不需要修改文件，shell调用一下就可以了

**Comment by Mister-Hope** [comment]
这样等我们发布大版本，就会有一堆小白过来开骂了。虽然说指定 v1 版本号比如 `"v1"` `">1.0.0 && < 2.0.0"` 可以解决 这个问题，但你仍不能保证是不是有用户会把lock文件一起上传。

另外你不能保证没有萌新会去因为环境变量的问题重新部署，所以让他们改版本号也许是一个最优选择，如果他们不会改，等用出毛病再改也不迟。

**Comment by lizheming** [comment]
「开启一下vercel的自动部署api，使用curl调用一下就更新到最新了。」 个人并不觉得 curl 调用一下比 GitHub 在线更改下版本号会简单一点

**Comment by appotry** [comment]
好吧，对于萌新可能修改配置文件更好。

但对于熟手来说，脚本自动化能大大提高效率。可以把博客所有的操作都用脚本控制起来，并且可以添加上各种控制选项

```
#!/bin/bash

echo -e "\033[32m[deblog2]\033[0m\033[32m:$LINENO\033[0m 部署vercel waline"
echo -e "\033[32m[deblog2]\033[0m\033[32m:$LINENO\033[0m waline.xxx.com"

curl -X POST https://api.vercel.com/v1/integrations/deploy/prj_xxxxxxxxxxxx7xEIwLG0ZfF/yxxxxx
```

**Reply by lizheming** [reply]
嗯嗯，没问题，多种方式都可以，满足自己的需求即可。
